### PR TITLE
pin-1858: Rate limiter

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala [hunk 1]
```diff
@@ -83,6 +76,20 @@ trait Dependencies {
   implicit val userRegistryApiKeyValue: UserRegistryApiKeyValue =
     userregistry.client.invoker.ApiKeyValue(ApplicationConfiguration.userRegistryApiKey)
 
+  val rateLimiter: RateLimiter =
+    RedisRateLimiter(ApplicationConfiguration.rateLimiterConfigs, OffsetDateTimeSupplierImpl)
+
+  val rateLimiterDirective: ExecutionContext => Seq[(String, String)] => Directive1[Seq[(String, String)]] = {
+    ec => contexts =>
+      {
+        val logger: LoggerTakingImplicit[ContextFieldsToLog] = Logger.takingImplicit[ContextFieldsToLog](this.getClass)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala]: not sure if this has downsides 🤔

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala]: What makes you doubt?

Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala]: Just the fact that we defined a logger in each class, and maybe it had some underlying reason that I'm missing

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala]: at least we don't instantiate a logger at each call

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala [hunk 2]
```diff
@@ -64,11 +70,13 @@ final case class AuthorizationApiServiceImpl(jwtReader: JWTReader, sessionTokenG
     claims.getClaims.asScala.view.filterKeys(admittedSessionClaims.contains).toMap
   }
 
-  private def getOrganizationId(claims: JWTClaimsSet): Try[AnyRef] = for {
-    nullableOrgClaimsMap <- Try(claims.getJSONObjectClaim(organizationClaim)).as(MissingClaim(organizationClaim))
+  private def getOrganizationId(claims: JWTClaimsSet): Try[UUID] = for {
+    nullableOrgClaimsMap <- Try(claims.getJSONObjectClaim(organizationClaim))
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala:78]: was it made on purpose? the semantic is pretty different, isn't it?

File: src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala (hunk 1 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/server/impl/dependencies/Dependencies.scala]: that was the purpose but I forgot to move the log creation 😂 
thanks for pointing it out, I'll fix it!

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala (hunk 2 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala:78]: `claims.getJSONObjectClaim` can return `Success(null)`, that's why I called the result `nullable`.
But feel free to suggest different naming!

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala [hunk 3]
```diff
@@ -12,18 +13,24 @@ import it.pagopa.interop.backendforfrontend.model.{IdentityToken, SessionToken}
 import it.pagopa.interop.commons.jwt.service.{JWTReader, SessionTokenGenerator}
 import it.pagopa.interop.commons.jwt.{getUserRoles, organizationClaim}
 import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.ratelimiter.RateLimiter
+import it.pagopa.interop.commons.ratelimiter.model.{Headers, RateLimitStatus}
 import it.pagopa.interop.commons.signer.model.SignatureAlgorithm
-import it.pagopa.interop.commons.utils.TypeConversions.{OptionOps, TryOps}
+import it.pagopa.interop.commons.utils.TypeConversions.{TryOps, _}
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala]: 🤔

Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala]: intellij stuff, I'll remove it

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala (hunk 2 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala:78]: The thing I'm saying is that AFAIK `as` is an alias of `map(_ => )`, while here you're using `.leftMap(_ => )`. I trust that the unit tests will have catched any strange behaviour though.

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala (hunk 2 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala:78]: I feel your confusion 🥲 
That `as` function has been defined in the commons under `TryOps` as an alias for `leftMap`, your point is exactly why I replaced it.
IMO we should remove that function from the commons

**Pull request merged**